test/perf/perf_test.go

```diff
@@ -95,3 +138,105 @@ func initLogger() (logr.Logger, *os.File, error) {
 	fmt.Printf("configured logger to write messages in '%s'\n", out.Name())
 	return logger, out, nil
 }
+
+// createMultipleSignups creates signups by batch (see `config.GetUserBatchSize()`) and monitors the CPU and memory while the
+// provisioning is in progress. Logs the max CPU and memory during captured during each batch by polling the `/metrics`
+// endpoint in a separate go routine.
+func createSignupsByBatch(t *testing.T, ctx *framework.Context, hostAwait *wait.HostAwaitility, memberAwait *wait.MemberAwaitility, config Configuration, logger logr.Logger) {
+
+	require.Equal(t, 0, config.GetUserCount()%config.GetUserBatchSize(), "number of accounts must be a multiple of %d", config.GetUserBatchSize())
+	hostMetricsRoute, err := hostAwait.SetupRouteForService("host-operator-metrics", "/metrics")
+	require.NoError(t, err)
+
+	t.Logf("provisionning %d accounts by batch of '%d", config.GetUserCount(), config.GetUserBatchSize())
+
+	hostOperatorPod, err := hostAwait.GetHostOperatorPod()
+	require.NoError(t, err)
+	memberOperatorPod, err := memberAwait.GetMemberOperatorPod()
+	require.NoError(t, err)
+
+	for b := 0; b < config.GetUserCount()/config.GetUserBatchSize(); b++ { // provisioning users by batch of `config.GetUserBatchSize()` (eg: 100)
+
+		signups := make([]toolchainv1alpha1.UserSignup, config.GetUserBatchSize())
+		start := time.Now()
+
+		for i := 0; i < config.GetUserBatchSize(); i++ {
+			n := (b*config.GetUserBatchSize() + i)
+			name := fmt.Sprintf("multiple-signup-testuser-%d", n)
+			// check if there is already a MUR with the expected name, in which case, continue with the next one
+			mur := toolchainv1alpha1.MasterUserRecord{}
+			if err := hostAwait.Client.Get(context.TODO(), types.NamespacedName{Namespace: hostAwait.Namespace, Name: name}, &mur); err == nil {
+				t.Logf("no need to create a UserSignup for '%s', the MasterUserRecord resource already exists", name)
+				// skip this one, it already exists
+				continue
+			}
+			// Create an approved UserSignup resource
+			userSignup := NewUserSignup(t, hostAwait, memberAwait, name, fmt.Sprintf("multiple-signup-testuser-%d@test.com", n), false)
+			userSignup.Spec.Approved = true
+			err := hostAwait.FrameworkClient.Create(context.TODO(), userSignup, CleanupOptions(ctx))
+			hostAwait.T.Logf("created usersignup with username: '%s' and resource name: '%s'", userSignup.Spec.Username, userSignup.Name)
+			require.NoError(t, err)
+			signups[i] = *userSignup
+		}
+		err := hostAwait.WaitUntilMetricHasValueOrLess(hostMetricsRoute.Status.Ingress[0].Host, "workqueue_depth", "name", "masteruserrecord-controller", 0)
+		require.NoError(t, err)
+
+		for _, signup := range signups {
+			mur, err := hostAwait.WaitForMasterUserRecord(signup.Spec.Username, UntilMasterUserRecordHasCondition(Provisioned()))
+			require.NoError(t, err)
+			// now, run a pod (with the `sleep 7200` command in each namespace)
```

Comment:
if you accept my following comment then it will be 28800 
```suggestion
			// now, run a pod (with the `sleep 28800` command in each namespace)
```